### PR TITLE
NAS-108687 / 20.12 / Save edited Apps Advanced Setting doesn't save

### PR DESCRIPTION
**Pull request by AlexKarpov98**

### CODE REVIEW

File: src/app/pages/common/entity/entity-form/entity-form.component.ts [hunk 1]
```diff
@@ -533,6 +539,12 @@ export class EntityFormComponent implements OnInit, OnDestroy, OnChanges, AfterV
                         (res) => {
                           this.loader.close();
                           this.loaderOpen = false;
+
+                          if (res.error) {
```
Comment (william-gr) [inline src/app/pages/common/entity/entity-form/entity-form.component.ts:543]: wonder if we need

`(res.editCallJob && res.error)`

?

File: src/app/pages/applications/forms/kubernetes-settings.component.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import { ApplicationsService } from '../applications.service';
 })
 export class KubernetesSettingsComponent {
   protected queryCall: string = 'kubernetes.config';
-  protected editCall: string = 'kubernetes.update';
+  protected editCallJob: string = 'kubernetes.update';
```
Comment (william-gr) [inline src/app/pages/applications/forms/kubernetes-settings.component.ts:16]: Not familiar with UI code but wonder if the job would make more sense to be a boolean so the editCall parameter is consistent everywhere

**Comment by dszidi** [comment]
Please remove fields/options relating to ipv6.

**Comment by william-gr** [comment]
Route ipv4 and interface are not required AFAIK

File: src/app/pages/common/entity/entity-form/entity-form.component.ts (hunk 1 above)
Comment (AlexKarpov98) [reply src/app/pages/common/entity/entity-form/entity-form.component.ts:543]: Thanks for your pointing out.

File: src/app/pages/applications/forms/kubernetes-settings.component.ts (hunk 2 above)
Comment (AlexKarpov98) [reply src/app/pages/applications/forms/kubernetes-settings.component.ts:16]: That make sence.

File: src/app/pages/applications/forms/kubernetes-settings.component.ts [hunk 3]
```diff
@@ -64,26 +70,15 @@ export class KubernetesSettingsComponent {
           name: 'route_v4_interface',
           placeholder: helptext.kubForm.route_v4_interface.placeholder,
           tooltip: helptext.kubForm.route_v4_interface.tooltip,
-          options: [{ label: '---', value: null}]
+          options: [{ label: '---', value: null}],
+          required: true,
```
Comment (william-gr) [inline src/app/pages/applications/forms/kubernetes-settings.component.ts:74]: Not required

File: src/app/pages/applications/forms/kubernetes-settings.component.ts [hunk 4]
```diff
@@ -64,26 +70,15 @@ export class KubernetesSettingsComponent {
...
-          placeholder: helptext.kubForm.route_v6_interface.placeholder,
-          tooltip: helptext.kubForm.route_v6_interface.tooltip,
-          options: [{ label: '---', value: null}]
-        },
-        {
-          type: 'input',
-          name: 'route_v6_gateway',
-          placeholder: helptext.kubForm.route_v6_gateway.placeholder,
-          tooltip: helptext.kubForm.route_v6_gateway.tooltip,
+          required: true,
```
Comment (william-gr) [inline src/app/pages/applications/forms/kubernetes-settings.component.ts:81]: Not required

File: src/app/pages/applications/forms/kubernetes-settings.component.ts [hunk 5]
```diff
@@ -106,24 +101,14 @@ export class KubernetesSettingsComponent {
...
 
   beforeSubmit(data) {
-    if (data.route_v4_gateway === '') {
-      data.route_v4_gateway = null;
-      console.log('yo', data.route_v4_gateway)
-    }
-    if (data.route_v6_gateway === '') {
-      data.route_v6_gateway = null;
-    }
+    data.node_ip = '0.0.0.0';
```
Comment (william-gr) [inline src/app/pages/applications/forms/kubernetes-settings.component.ts:112]: Not sure why we made this changed? Someone requested?

Comment (AlexKarpov98) [reply src/app/pages/applications/forms/kubernetes-settings.component.ts]: yes it was, but reverted.

File: src/app/pages/applications/forms/kubernetes-settings.component.ts (hunk 4 above)
Comment (AlexKarpov98) [reply src/app/pages/applications/forms/kubernetes-settings.component.ts]: removed it

File: src/app/pages/applications/forms/kubernetes-settings.component.ts (hunk 3 above)
Comment (AlexKarpov98) [reply src/app/pages/applications/forms/kubernetes-settings.component.ts]: Removed it

File: src/app/pages/applications/forms/kubernetes-settings.component.ts [hunk 6]
```diff
@@ -106,24 +98,13 @@ export class KubernetesSettingsComponent {
...
       interfaces.forEach(i => {
         v4_interface_control.options.push({ label: i.name, value: i.name });
-        v6_interface_control.options.push({ label: i.name, value: i.name });
       })
     })
-
   }
 
   beforeSubmit(data) {
-    if (data.route_v4_gateway === '') {
```
Comment (william-gr) [inline src/app/pages/applications/forms/kubernetes-settings.component.ts:120]: Did we had to remove this for route v4? Or just wasnt necessary?

**Review by dszidi — APPROVED** [review]
LGTM

File: src/app/pages/applications/forms/kubernetes-settings.component.ts (hunk 6 above)
Comment (sonicaj) [reply src/app/pages/applications/forms/kubernetes-settings.component.ts:120]: I am not sure how this works, but just to be clear, we need to send `null` to allow user to unset this field.

**Review by sonicaj — APPROVED** [review]

**Pull request merged**